precise/iaf_psc_exp_ps_time_reversal.cpp

```diff
@@ -0,0 +1,720 @@
+  
+  if ( tau_m_ <= 0 || tau_ex_ <= 0 || tau_in_ <= 0 )
+    throw BadProperty("All time constants must be strictly positive.");
+
+   if ( tau_m_ == tau_ex_ || tau_m_ == tau_in_ )
+    throw BadProperty("Membrane and synapse time constant(s) must differ."
+                      "See note in documentation.");
+
+   calc_const_spike_test_();
+
+   return delta_EL;
+}
+
+void nest::iaf_psc_exp_ps_time_reversal::State_::get(DictionaryDatum & d, 
+                                       const Parameters_ & p) const
+{
+  def<double>(d, names::V_m, y2_ + p.E_L_);  // Membrane potential
+  def<bool>(d, names::is_refractory, is_refractory_);
+  def<double>(d, names::t_spike, Time(Time::step(last_spike_step_)).get_ms());
+  def<double>(d, names::offset, last_spike_offset_);
+  def<double>(d, names::y1_ex, y1_ex_);
+  def<double>(d, names::y1_in, y1_in_); // y1 state
+  def<double>(d, names::y2, y2_); // y2 state
+  def<double>(d, names::I_syn, y1_ex_ + y1_in_); 
+
+
+  // these entries will break ticket-459
+  // because they chane depending on E_L (which is correct)
+  // since they are only used for debugging we will comment them
+  // out for the time being
+  //def<long>(d, names::pot_spikes, c0);
+  //def<double>(d, names::dhaene_quick1, (dhaene_quick1*100.0/c0));
+  //def<double>(d, names::dhaene_quick2, (dhaene_quick2*100.0/c0));
+  //def<double>(d, names::dhaene_tmax_lt_t1, (dhaene_tmax_lt_t1*100.0/c0));
+  //def<double>(d, names::dhaene_max_geq_V_th, (dhaene_max*100.0/c0));
+  //def<double>(d, names::dhaene_det_spikes, (dhaene_det_spikes*100.0/c0));
+
+  //def<double>(d, names::eq7, (c1a*100.0/c0));
+  //def<double>(d, names::eq9, (c1b*100.0/c0));
+  //def<double>(d, names::eqs7and9, (c2*100.0/c0));
+  //def<double>(d, names::lin_left_geq_V_th, (c3a*100.0/c0));
+  //def<double>(d, names::lin_max_geq_V_th, (c3b*100.0/c0));
+  //def<double>(d, names::eq13, (c4*100.0/c0));
+  //def<double>(d, names::eq12, (det_spikes*100.0/c0));
+}
+
+void nest::iaf_psc_exp_ps_time_reversal::State_::set(const DictionaryDatum & d, const Parameters_ & p, double delta_EL)
+{
+  if ( updateValue<double>(d, names::V_m, y2_) )
+    y2_ -= p.E_L_;
+  else
+    y2_ -= delta_EL;
+
+  //double_t dv;
+  //updateValue<double>(d, names::dhaene_quick1, dv);
+  //updateValue<double>(d, names::dhaene_quick2, dv);
+  //updateValue<double>(d, names::dhaene_tmax_lt_t1, dv);
+  //updateValue<double>(d, names::dhaene_max_geq_V_th, dv);
+  //updateValue<double>(d, names::dhaene_det_spikes, dv);
+  //updateValue<double>(d, names::eq7, dv);
+  //updateValue<double>(d, names::eq9, dv);
+  //updateValue<double>(d, names::eqs7and9, dv);
+  //updateValue<double>(d, names::lin_left_geq_V_th, dv);
+  //updateValue<double>(d, names::lin_max_geq_V_th, dv);
+  //updateValue<double>(d, names::eq13, dv);
+  //updateValue<double>(d, names::eq12, dv);
+  updateValue<double>(d, names::y1_ex, y1_ex_ );
+  updateValue<double>(d, names::y1_in, y1_in_);
+  updateValue<double>(d, names::y0, y0_);
```

Comment:
`y0_` is the current injected through current-transmitting synapses. It should not be settable.